### PR TITLE
lollypop-portal: init at 0.9.7

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
This #43292 

###### Notes
I didn't exactly know where this belonged in nixpkgs so feel free to suggest a better place.
There will also need to be a stupid simple nixos module for this.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
@cocreature Can you test this?

File: pkgs/misc/lollypop-portal/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    pydbus
+    pycairo
+  ];
+
+  dontWrapGApps = true;
+
+  postFixup = ''
+    buildPythonPath "$out/libexec/lollypop-portal $pythonPath"
+
+    wrapProgram $out/libexec/lollypop-portal \
```
Comment (jtojnar) [inline pkgs/misc/lollypop-portal/default.nix]: Alternately, you can append to `gappsWrapperArgs`: https://github.com/NixOS/nixpkgs/blob/9bbb84f622e8c421419e6ca9949a5b55e25c4be8/pkgs/desktops/gnome-3/core/nautilus/default.nix#L26-L33

Comment (worldofpeace) [reply pkgs/misc/lollypop-portal/default.nix]: Lol That's what I was trying to do

Comment (worldofpeace) [reply pkgs/misc/lollypop-portal/default.nix]: It seems that this won't make a wrapper for the program at libexec which is why I think I did this.

Comment (jtojnar) [reply pkgs/misc/lollypop-portal/default.nix]: Works for me:

```diff
--- a/pkgs/misc/lollypop-portal/default.nix
+++ b/pkgs/misc/lollypop-portal/default.nix
@@ -14,7 +14,7 @@
      sha256 = "0rn5xmh6391i9l69y613pjad3pzdilskr2xjfcir4vpk8wprvph3";
   };
 
-  nativeBuildInputs = [ 
+  nativeBuildInputs = [
     gobjectIntrospection
     meson
     ninja
@@ -41,15 +41,12 @@
     pycairo
   ];
 
-  dontWrapGApps = true;
-
-  postFixup = ''
+  preFixup = ''
     buildPythonPath "$out/libexec/lollypop-portal $pythonPath"
-
-    wrapProgram $out/libexec/lollypop-portal \
-      "''${gappsWrapperArgs[@]}" \
-        --prefix PYTHONPATH : "$program_PYTHONPATH" \
-        --prefix PATH : "${stdenv.lib.makeBinPath [ easytag kid3 ]}"
+    gappsWrapperArgs+=(
+      --prefix PYTHONPATH : "$program_PYTHONPATH"
+      --prefix PATH : "${stdenv.lib.makeBinPath [ easytag kid3 ]}"
+    )
   '';
 
   meta = with stdenv.lib; {
```

Comment (worldofpeace) [reply pkgs/misc/lollypop-portal/default.nix]: Weird I ended up with the same change and it fails?

Comment (jtojnar) [reply pkgs/misc/lollypop-portal/default.nix]: Yep, produces the same wrapper file.

**Comment by cocreature** [comment]
@worldofpeace Thanks a lot! I can confirm that this works for me!

I have two small comments:
1. It might make sense to make `lollypop` depend on this. Technically, it is not a required dependency but I am not sure if this is intended by upstream. At the very least a comment in the `lollypop` derivation might be useful since just not having menu items show up is somewhat confusing.
2. You seem to have added `kid3` and `easytag` to the PATH of `lollypop-portal`. I wonder if it might not be best to just have users install those packages if they want them especially given that `lollypop` supports more tag editors (see https://gitlab.gnome.org/World/lollypop/blob/master/lollypop/define.py#L24 but I think you could even configure an arbitrary editor).

**Comment by worldofpeace** [comment]
@cocreature Thanks for your comments.

> It might make sense to make lollypop depend on this. Technically, it is not a required dependency but I am not sure if this is intended by upstream. At the very least a comment in the lollypop derivation might be useful since just not having menu items show up is somewhat confusing.

It is true that this isn't a required dependency because the author is responsible for the ubuntu packaging and it's not required there. Maybe I could improve the package descriptions also, since a `nix search lollypop` would yield both of these packages. IMHO the author doesn't make this very clear in the first place.

> You seem to have added kid3 and easytag to the PATH of lollypop-portal. I wonder if it might not be best to just have users install those packages if they want them especially given that lollypop supports more tag editors (see https://gitlab.gnome.org/World/lollypop/blob/master/lollypop/define.py#L24 but I think you could even configure an arbitrary editor).

kid3-cli needs to be in the path for you to set the cover and popularity.
So that would obviously belong in the lollypop-portal derivation default.
As for all the various tag editors, Perhaps we could utilize a nixos module for this configuration or options in the derivation(which are hard to discover).

**Comment by cocreature** [comment]
Alright, sounds good! No further comments from my side. For now just depending on easytag is totally fine for me.

**Comment by worldofpeace** [comment]
Your exclamation confused me. :laughing:

**Comment by peti** [comment]
This commit introduces evaluation errors into Nixpkgs:

~~~
$ nix-env -qaP --meta --json  >/dev/null
derivation 'lollypop-portal' has invalid meta attribute 'override'
derivation 'lollypop-portal' has invalid meta attribute 'overrideDerivation'
~~~

**Comment by worldofpeace** [comment]
@peti umm any idea why?

**Comment by worldofpeace** [comment]
@peti This [commit](https://github.com/NixOS/nixpkgs/commit/aab3182ca4a2b69bfd01db53e1e95c941425203b) fixed that. I frequently misspell that.

**Comment by peti** [comment]
Cool, thank you!

**Pull request merged**